### ISSUE TITLE
Add running timer notification

### ISSUE THREAD

**Issue by amulware** [toggl/mobileapp#3055]
This is an optional feature that can be enabled in the settings. When it is enabled, as long as the user is logged in and has a running timer we should show a permanent notification that will inform the user that the timer is running.

This notification shows data from the time entry, including a real-time duration. It allows the user to go straight back to the app, and may also include a stop button.

If the user changes the appropriate setting, the notification should appear/disappear right away if needed.

**Comment by pe0ny** [comment]
<img width="369" alt="screen shot 2018-08-24 at 09 56 56" src="https://user-images.githubusercontent.com/5064445/44569755-1256cc00-a784-11e8-8068-5e8456fe0e72.png">

### The header area is comprised of:

**1. The app icon:**
The app icon that appears in monochrome in the status bar.

**2. The app name:**
The name of the app automatically appears in the notification.

**3. Header text:**
That's where we would like to display running time entry's duration.

### The content area:

**1. A content title:**
That's where we add time entry's description. If the time entry doesn't have a description, just display: "No description".

**2. Content text:**
That's where we display project, task and client. Since we can't play with colours there (as far as i know), the format should be the following:
Project: task (Client)
💡 If it's possible, the better way to display these would be like this (with the middle dot, instead of brackets): Project: task · client
If the time entry doesn't have a project, just display: "No project"

Tapping the notification itself will open the app in the Timer page.

Expandable option will be added in the future but not in this phase.

**Comment by daividssilverio** [comment]
Hey @pe0ny , about the header text. That part of the notification is system managed, and not easily customizable (it is possible, but we have guarantee it won't break on different devices that might apply different styles to the notification area; and we won't match the rest of the notifications on the notification area).

My point is: can we have just the :toggl: "09:10" without the "·" for the notification header? We can have the system update the notifications TE's timer automatically based on the TE start time.

**Comment by daividssilverio** [comment]
Another point: how far are we from the possibility of the backend sending us push notifications informing the up about new running time entries/stopped time entries?

That would make it so much easier to remove the notification when the user is online but stops the current notification from the web app.

We could just rely on syncs inside the app or pull from the server periodically, but the former depends on the user opening the app and syncing and the later can make the feature a battery hog.

**Comment by kornelijepetak** [comment]
I don't think we should change the default Google design of notifications. Just sayin'. I've had bad experience with that.

**Comment by daividssilverio** [comment]
Can we have the notification icon exported?

**Comment by pe0ny** [comment]
@daividssilverio 

> My point is: can we have just the :toggl: "09:10" without the "·" for the notification header? We can have the system update the notifications TE's timer automatically based on the TE start time.

Yes. Let's not add "·" then.

> how far are we from the possibility of the backend sending us push notifications informing the up about new running time entries/stopped time entries?

This won't happen before we drop the old app for good.

> Can we have the notification icon exported?

Yes. https://zpl.io/aMvn7DN

**Comment by pe0ny** [comment]
@kornelijepetak 

> I don't think we should change the default Google design of notifications. Just sayin'. I've had bad experience with that.

What do you mean exactly?

**Comment by daividssilverio** [comment]
Hey @pe0ny, are we gonna do something about big description/project/task names?
The basic notification layout will cut off the exceeding text.
If the project name is too big, the task and client might not have space to show.
If the task name is too big, the client might not have space to show.

We have some options:
1. Leave as it is; do something later
2. Abbreviate the project/task/client names if they exceed some length (I have to stress this, but an actual good length is tricky to get... devices densities vary a lot)
3. We can leave the current title and content as they are, and add the full details of the TE in an expansible area, as shown here: https://developer.android.com/training/notify-user/expanded
4. Create a custom layout for the notification body (not the headers, the content area) and handle it there. I rather not follow this path.

**Comment by pe0ny** [comment]
@daividssilverio Let's go with the option 1 for now. We'll have expandable notification later, so then we can reveal the whole text.
